seatunnel-connectors-v2/connector-influxdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/influxdb/source/InfluxdbSourceReader.java

```diff
@@ -0,0 +1,150 @@
+import org.apache.seatunnel.connectors.seatunnel.influxdb.converter.InfluxDBRowConverter;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+import org.influxdb.InfluxDB;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+
+import java.io.IOException;
+import java.net.ConnectException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+@Slf4j
+public class InfluxdbSourceReader implements SourceReader<SeaTunnelRow, InfluxDBSourceSplit> {
+    private InfluxDB influxDB;
+    InfluxDBConfig config;
+
+    private static final long THREAD_WAIT_TIME = 500L;
+
+    private Set<InfluxDBSourceSplit> sourceSplits;
+
+    private final Context context;
+
+    private SeaTunnelRowType seaTunnelRowType;
+
+    List<Integer> columnsIndexList;
+
+    InfluxdbSourceReader(InfluxDBConfig config, Context readerContext, SeaTunnelRowType seaTunnelRowType, List<Integer> columnsIndexList) {
+        this.config = config;
+        this.sourceSplits = new HashSet<>();
+        this.context = readerContext;
+        this.seaTunnelRowType = seaTunnelRowType;
+        this.columnsIndexList = columnsIndexList;
+    }
+
+    public void connect() throws ConnectException {
+        if (influxDB == null) {
+            influxDB = InfluxDBClient.getInfluxDB(config);
+            String version = influxDB.version();
+            if (!influxDB.ping().isGood()) {
+                String errorMessage =
+                        String.format(
+                                "connect influxdb failed, due to influxdb version info is unknown, the url is: {%s}",
+                                config.getUrl());
+                throw new ConnectException(errorMessage);
+            }
+            log.info("connect influxdb successful. sever version :{}.", version);
+        }
+    }
+
+    @Override
+    public void open() throws Exception {
+        connect();
+    }
+
+    @Override
+    public void close() throws IOException {
+        if (influxDB != null) {
+            influxDB.close();
+            influxDB = null;
+        }
+    }
+
+    @Override
+    public void pollNext(Collector<SeaTunnelRow> output) throws Exception {
+        if (sourceSplits.isEmpty()) {
+            Thread.sleep(THREAD_WAIT_TIME);
+            return;
+        }
+        sourceSplits.forEach(source -> {
+            try {
+                read(source, output);
+            } catch (Exception e) {
+                throw new RuntimeException("influxDB source read error", e);
+            }
+        });
+
+        if (Boundedness.BOUNDED.equals(context.getBoundedness())) {
+            // signal to the source that we have reached the end of the data.
+            log.info("Closed the bounded fake source");
+            context.signalNoMoreElement();
+        }
+    }
+
+    @Override
+    public List<InfluxDBSourceSplit> snapshotState(long checkpointId) throws Exception {
+        return null;
```

Comment:
Please add split snapshot.
you can see https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSourceReader.java

Reply:
> Please add split snapshot. you can see https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-connectors-v2/connector-iotdb/src/main/java/org/apache/seatunnel/connectors/seatunnel/iotdb/source/IoTDBSourceReader.java

already fixed